lightning/src/ln/msgs.rs

```diff
@@ -826,6 +826,33 @@ pub trait RoutingMessageHandler : Send + Sync {
 	fn should_request_full_sync(&self, node_id: &PublicKey) -> bool;
 }
 
+
+/// A trait that describes an object that can receive gossip queries messages
```

Comment:
I think we should just integrate this into RoutingMessageHandler. Indeed, it would make RoutingMessageHandler much more complex, but that's the way the protocol went, and the existing functions are not very useful without this, too. That would also create significantly less conflicts with other work, as Julian notes.

Reply:
Sounds good, I've been thinking the same thing. Thanks!